lib/src/utils.dart

```diff
@@ -426,6 +426,26 @@ bool get canUseAnsiCodes {
 /// Gets an ANSI escape if those are supported by stdout (or nothing).
 String getAnsi(String ansiCode) => canUseAnsiCodes ? ansiCode : '';
 
+/// Whether an environment variable overriding the [stdout.hasTerminal] check
+/// was passed.
+bool get forceTerminalOutput =>
+    Platform.environment.containsKey('_PUB_FORCE_TERMINAL_OUTPUT');
```

Comment:
Alternative env var names:
- `_PUB_OVERRIDE_HAS_TERMINAL_CHECK`
- `_PUB_OVERRIDE_TERMINAL_CHECK`
- `_PUB_OVERRIDE_STDOUT_HAS_TERMINAL`

---

Also, we could use one of these existing flags from lib/command_runner.dart instead of a new env var. I'm not sure if there would be side effects for commands other than `pub get`, though.
https://github.com/dart-lang/pub/blob/97f638d091e1afa66f422d0b83f71a0d65692b4e/lib/src/command_runner.dart#L39-L44

Reply:
We could also consider reusing the mechanism we have for deciding whether to show colors or not - I guess showing an animation could fall under the same umbrella...

Reply:
True, yea I can take a look.